ledger/accountdb.go

```diff
@@ -1989,6 +1996,39 @@ func (qs *accountsDbQueries) lookupKeyValue(key string) (pv persistedValue, err
 	return
 }
 
+func (qs *accountsDbQueries) lookupKeysByPrefix(prefix string, maxKeyNum uint64, results map[string]bool, resultCount uint64) (round basics.Round, err error) {
+	err = db.Retry(func() error {
+		var rows *sql.Rows
+		// Cast to []byte to avoid interpretation as character string, see note in upsertKvPair
+		rows, err = qs.lookupKeysByPrefixStmt.Query(len(prefix), []byte(prefix))
```

Comment:
I think you _do_ want (or at least it's ok) to create a new `err` here.  The outer `err` will get set because you return the inner `err` from the inner function. See other lookupAllResources for a similar example.

```suggestion
		// Cast to []byte to avoid interpretation as character string, see note in upsertKvPair
		rows, err := qs.lookupKeysByPrefixStmt.Query(len(prefix), []byte(prefix))
```